### DISCUSSION TITLE
Help for deploying a MEAN stack

### DISCUSSION THREAD

**Discussion by eliagentili** [caprover/caprover#1081] [Q&A]
Hi everyone,
I'm not a Docker expert, and I need some help to deploy my MEAN stack.

Everything works well on Heroku, but I can't find the right solution for CapRover.

[Here you can see an example repo](https://github.com/eliagentili/basic-login-app)

I try to change my Docker file in many ways, but what works pretty straightforward with Heroku it doesn't here.

Please, can someone help me to identify where I'm wrong?

Thank you, everybody.

**Comment by githubsaturn** [comment]
Try this Dockerfile. You don't need nginx as your backend app takes care of [serving static content](https://github.com/eliagentili/basic-login-app/blob/main/server/src/config/init/prod.js#L51-L57).

Having said that, your application build process is **extremely slow**... I suggest you use a separate machine for your build. [Read here](https://caprover.com/docs/ci-cd-integration.html). 

```
FROM node:14-alpine
RUN apk update && apk upgrade && \
    apk add --no-cache git

RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app

COPY ./ /usr/src/app
RUN npm install --production && npm cache clean --force
RUN npm run build
ENV NODE_ENV production
ENV PORT 80
EXPOSE 80

CMD [ "npm", "start" ]

```


I tested this and it works fine:
![image](https://user-images.githubusercontent.com/10582082/113871645-1c853780-9781-11eb-9470-87ec4bde9297.png)

**Reply by eliagentili** [reply]
I can't understand why it doesn't work for me.

I retry also changing the engines in the `package.json` file with node 14.16.1.

I leave here a link to the logs:

[logs file](https://wormhole.app/wR6ql#cFiyKK0VjDKqy9-iY6ivqg)

**Reply by githubsaturn** [reply]
Did you use the exact same dockerfile that I posted above? Did you make sure you committed the change to your git?

**Reply by eliagentili** [reply]
Yes, for sure, you can see the repo here

[Repo with changes](https://github.com/eliagentili/basic-login-app/blob/main/Dockerfile)

**Reply by githubsaturn** [reply]
The only reason is that your build consumes too much ram and your machine can't handle it. I tested it on an 8gb machine.

**Reply by eliagentili** [reply]
Yes, probably.

I tested on a basic droplet, and I also see that there were some problems on DigitalOcean.

[DO status](https://status.digitalocean.com/)

I retry upgrading my droplet and let you know 😉 

Thanks for your help.

**Reply by eliagentili** [reply]
And now it works pretty well, and it's super fast.

I supposed that the problem was on DigalOcean's side.

Thanks
